### PR TITLE
fix: update migration test & update gossip endpoint to respect existing address mode during restart and upgrade

### LINKED ISSUE #4267: PR Check migration test with solo 0.72.0 was failed
### Describe the bug

https://github.com/hiero-ledger/solo/actions/runs/25384686119/job/74447609967

This is due to a recent PR change, gossip endpoint could be cluster ip address instead of FQCN address

### Describe the expected behavior

Should not fail

### To Reproduce

PR Checks

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by JeffreyDallas**
## Description

This pull request changes the following:

* If application.property `node.gossipFqdnRestricted` is set to true, address for gossip endpoint should be cluster ip, otherwise it could be FQCN address
* During restart from saved state, solo should respect the endpoint address saved in saved state either it is cluster i p address or FQCN address, instead of generating new ones.
* Challenging part is during migration, restart helm chart cause the cluster ip could be changed, so migration script need to track old ip address and patch helm chart accordingly after upgrade.

### Related Issues

* Closes #4267
* Related to #

### Pull request (PR) checklist

* \[x] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[x] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[x] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[x] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

**Comment by JeffreyDallas** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/hiero-ledger/solo/pull/4266#pullrequestreview-4240996796)

File: src/core/profile-manager.ts [hunk 1]
```diff
@@ -758,4 +867,49 @@ export class ProfileManager {
...
+    );
+    if (match?.[1]) {
+      return match[1].toLowerCase() === 'true';
+    }
+    return undefined;
+  }
+
+  private async getGossipFqdnRestricted(
+    consensusNodes: ConsensusNode[],
+    applicationPropertiesPath: string,
```
[the remark this answers is not in the document]
Comment (JeffreyDallas) [reply src/core/profile-manager.ts:883]: @copilot apply changes based on [this feedback](https://github.com/hiero-ledger/solo/pull/4266#discussion_r3198981056)

File: src/core/profile-manager.ts [hunk 2]
```diff
@@ -675,6 +686,96 @@ export class ProfileManager {
...
+        this.logger.info(`Found saved endpoint for ${consensusNode.name}: ${ipAddress}:${port} (IP)`);
+        return new Address(port, ipAddress);
+      }
+
+      return undefined;
+    } catch (error: Error | unknown) {
+      // If anything fails, return undefined to fall back to getExternalAddress
+      this.logger.debug(
+        `Could not extract saved endpoint for ${consensusNode.name}: ${error instanceof Error ? error.message : 'unknown error'}`,
+      );
```
Comment (jan-milenkov) [inline src/core/profile-manager.ts:774]:
```suggestion
    } catch (error) {
      // If anything fails, return undefined to fall back to getExternalAddress
      this.logger.debug(
        `Could not extract saved endpoint for ${consensusNode.name}: ${error.message}`,
      );
```

**Review by jan-milenkov — APPROVED** [review]

File: src/core/profile-manager.ts (hunk 2 above)
Comment (jan-milenkov) [reply src/core/profile-manager.ts:774]: optional change
